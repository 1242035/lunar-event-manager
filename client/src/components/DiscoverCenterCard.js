import React from 'react';
import placeholder from '../assets/images/placeholder.jpg';

/**
 * Component in the shape of a card that contains brief details about a center.
 * @param{Object} props - passed properties
 * @return{React.Component}
 */
const DiscoverCenterCard = (props) => {
  const {
    name, type, location, description,
  } = props;
  let { imageUrl } = props;

  // If image is not provided, use a placeholder image
  if (!imageUrl) {
    imageUrl = placeholder;
  }

  return (
    <div className="io-center-card animated zoom-in" >
      <div className="io-img"><img alt="" src={imageUrl} /></div>
      <div className="io-content">
        <div className="io-title">{name}</div>
        <div className="io-body">{type}</div>
        <hr />
        <p className="description">{description}</p>
        <p>{location}</p>
      </div>
    </div>
  );
};

export default DiscoverCenterCard;
