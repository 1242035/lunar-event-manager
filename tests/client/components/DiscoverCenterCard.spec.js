import React from 'react';
import { shallow } from 'enzyme';
import DiscoverCenterCard from '../../../client/src/components/DiscoverCenterCard';

describe('ConfirmModal', () => {
  let wrapper;
  const props = {
    name: '',
    type: '',
    location: '',
    description: '',
  };

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(
      <DiscoverCenterCard
        name={props.name}
        type={props.type}
        location={props.location}
        description={props.description}
      />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render connected component', () => {
    wrapper = shallow(
      <DiscoverCenterCard
        name={props.name}
        type={props.type}
        location={props.location}
        description={props.description}
      />);
    expect(wrapper.length).toEqual(1);
  });
});
