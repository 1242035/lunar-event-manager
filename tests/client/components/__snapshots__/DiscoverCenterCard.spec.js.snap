// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DiscoverCenterCard
    description=""
    location=""
    name=""
    type=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="io-img"
        >
          <img
            alt=""
            src="test-file-stub"
          />
        </div>,
        <div
          className="io-content"
        >
          <div
            className="io-title"
          >
            
          </div>
          <div
            className="io-body"
          >
            
          </div>
          <hr />
          <p
            className="description"
          >
            
          </p>
          <p>
            
          </p>
        </div>,
      ],
      "className": "io-center-card animated zoom-in",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt=""
            src="test-file-stub"
          />,
          "className": "io-img",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "test-file-stub",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="io-title"
            >
              
            </div>,
            <div
              className="io-body"
            >
              
            </div>,
            <hr />,
            <p
              className="description"
            >
              
            </p>,
            <p>
              
            </p>,
          ],
          "className": "io-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "io-title",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "io-body",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "description",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="io-img"
          >
            <img
              alt=""
              src="test-file-stub"
            />
          </div>,
          <div
            className="io-content"
          >
            <div
              className="io-title"
            >
              
            </div>
            <div
              className="io-body"
            >
              
            </div>
            <hr />
            <p
              className="description"
            >
              
            </p>
            <p>
              
            </p>
          </div>,
        ],
        "className": "io-center-card animated zoom-in",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
              src="test-file-stub"
            />,
            "className": "io-img",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "test-file-stub",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="io-title"
              >
                
              </div>,
              <div
                className="io-body"
              >
                
              </div>,
              <hr />,
              <p
                className="description"
              >
                
              </p>,
              <p>
                
              </p>,
            ],
            "className": "io-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "io-title",
              },
              "ref": null,
              "rendered": "",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "io-body",
              },
              "ref": null,
              "rendered": "",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "description",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
